### ISSUE TITLE
Project goals?

### ISSUE THREAD

**Issue by pickfire** [helix-editor/helix#65]
I wonder what are the project goals of helix. I have used quite a few editors, vim, neovim, vis, kakoune, doom emacs but I wonder what are the goals of this project? All of them have some stuff that I like but up till now all have their own downsides. I wonder what helix aims to achieves.

Also, since the project is new, we have the luxury to change the keymaps at the start, having used quite a few editors I think some of the keys in kakoune could be improved (some not easy to remember and some useless, like `x` vs `X`). Maybe we could create an RFC and discuss the alternatives and which keys to do what? By the way, I like kakoune alt, but not the limitation of terminal not being able to differentiate between ctrl-shift and ctrl, alt is similar case IIRC.

**Comment by rieger-jared** [comment]
I guess this is also associated with the issue https://github.com/helix-editor/helix/issues/165

I would also like to contribute to what I hope the project would become. 

I've also used Doom emacs, nvim, vscode but like you've said @pickfire, they all have their own downsides. For example, I found Doom emacs to be slow and too bulky, however I liked that it came preconfigured with the ability to perform file navigation, search ect. nvim is great in the fact that it's really customisable. I also find that this is also it's negative, as a lot of the plugins I end up installing are essential features that I find are missing in an editor. For example, tpopes "surround" plugin that allows a user to manipulate quoted text. Or telescope that allows a user to use fzf finding to quickly access files. 

I guess it also depends on how opinionated the editor wants to be.
